_posts/pl/2021-02-10-przyklad-migracji-do-chmury.md

```diff
@@ -0,0 +1,888 @@
+
+- konfiguracji tworzącej buckety w Cloud Storage (**infrastructure/frontend-bucket.tf** i **infrastructure/yarn-cache.storage.tf**), które wykorzystywane są m.in. do przechowywania plików statycznych aplikacji frontendowej oraz kopii podręcznej zależności, wykorzystywanych przez aplikację frontendową i funkcję (w dalszej części wyjaśnię o co chodzi),
+- konfiguracji topicu w usłudze Cloud Pub/Sub (**infrastructure/pubsub.tf**),
+- konfiguracji triggerów w usłudze Cloud Build, które na podstawie zmian w repozytorium GitHub, uruchamiają zdefiniowane zadania (o tym też za chwilę).
+
+### Konfiguracja Cloud Build
+
+Drugą z konfiguracji, jest konfiguracja usługi Cloud Build, czyli definicja wykonania zadań, potrzebnych do zbudowania i wdrożenia poszczególnych elementów systemu. Definicja wdrażania dla każdego elementu systemu znajduje się w folderze zawierającym jego kody źródłowe. Składa się ona z:
+
+- konfiguracji wdrażania infrastruktury (**infrastructure/build/deploy.cloudbuild.yml**):
+
+```yaml
+steps:
+  - name: 'hashicorp/terraform:0.12.26'
+    entrypoint: 'sh'
+    args:
+      - '-c'
+      - |
+        terraform init && terraform apply -auto-approve
+    dir: 'infrastructure'
+```
+
+Składa się ona z jednego kroku, który za pomocą powłoki systemu Linux (sh), wykonuje w folderze **infrastructure** następujące polecenia: **terraform init** (inicjalizujące Terraform), **terraform apply -auto-approve** (tworzące infrastrukturę za pomocą konfiguracji Terraform),
+
+- konfiguracji budowania i wdrażania aplikacji frontendowej (**frontend/deploy.cloudbuild.yml**),
+- konfiguracji wdrażania funkcji (**function/deploy.cloudbuild.yml**).
+
+### Dependency cache
+
+Kopia podręczna zależności, wykorzystywanych przez aplikację frontendową oraz funkcję, przechowywana jest w przeznaczonym do tego celu buckecie.
+
+Aplikacje JavaScript (a więc te oparte o framework Angular, jak i funkcje napisane w czystym JavaScript) wykorzystują zewnętrzne zależności (biblioteki, moduły), które przechowują w folderze **node_modules**. Za każdym razem, kiedy aplikacja lub funkcja są budowane, muszą one pobrać wszystkie zależności z których korzystają. Bardzo często, zależności liczone są w dziesiątkach, a nawet setkach, a pobranie wszystkich za każdym razem, powoduje niepotrzebne wykorzystywanie łącza internetowego oraz dłuższy czas wykonywania zadania w usłudze Cloud Build (która posiada darmowy limit w postaci 120 minut, za które w ciągu dnia nie trzeba dodatkowo płacić).
+
+Chcąc zoptymalizować czas budowania i wdrażania, a także koszty z tym związane, utworzone zostały skrypty (w folderze **yarn-cache-builder**), które przed pobraniem zależności z Internetu, sprawdzają najpierw, czy nie istnieje już paczka (plik .zip), zawierające te zależności w Cloud Storage. Jeśli tak, to zależności te pobierane są z bucketu i wypakowywane, co zajmuje o wiele mniej czasu, niż pobieranie ich ponownie.
```

Comment:
odnośnie do "zawierające te zależności w Cloud Storage."
to podmiotem są skrypty czy paczka? Bo wychodzi z konstrukcji zdania, że paczka, a wówczas trzeba by zmienić "na zawierająca".

Reply:
miało być "zawierająca"